osu.Game/IO/StableStorage.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.IO;
+using System.Linq;
+using osu.Framework.Platform;
+
+namespace osu.Game.IO
+{
+    /// <summary>
+    /// A storage pointing to an osu-stable installation.
+    /// Provides methods for handling installations with a custom Song folder location.
+    /// </summary>
+    public class StableStorage : DesktopStorage
+    {
+        private const string stable_songs_path = "Songs";
+
+        private readonly DesktopGameHost host;
+        private readonly string songsPath;
+
+        public StableStorage(string path, DesktopGameHost host)
+            : base(path, host)
+        {
+            this.host = host;
+            songsPath = locateSongsDirectory();
+        }
+
+        /// <summary>
+        /// Returns a <see cref="Storage"/> pointing to the osu-stable Songs directory.
+        /// </summary>
+        public Storage GetSongStorage() => new DesktopStorage(songsPath, host);
+
+        private string locateSongsDirectory()
+        {
+            var configFile = GetStream(GetFiles(".", "osu!.*.cfg").First());
```

Comment:
can this match multiple files? seems like this'll pick one at random?

also why isn't this stream usage in a `using`? isn't this leaving the cleanup of the file stream to the finaliser?

Reply:
Should also consider that this could throw if there happens to be no config file. Probably not what we want (if someone has a `Songs` folder but deleted their config, it shouldn't fail the import from stable. with your changes it will).

Reply:
> can this match multiple files? seems like this'll pick one at random?

This should pickup the first osu! user config file available (containing the desired options) if it exists, according to the pattern passed in.

Reply:
> This should pickup the first osu! user config file available (containing the desired options) if it exists, according to the pattern passed in.

Yeah I can read code, thanks. Restating what the code does not say anything about if it's correct, which is what I'm questioning.

Reply:
AFAICT, the osu! user config file contains the local session username in its filename even though stable installs in `%localappdata%` which isn't shared between sessions so there shouldn't be multiple user config files.

Reply:
I could see an edge case where a user changes their windows username via the control panel, at which point their app data folder gets auto-migrated, but the filename is not updated (because how would windows know to update it) and there are now potentially two files. I would probably make this a `Single()` just so that *some* error is raised.